error.go

```diff
@@ -45,3 +46,74 @@ func NewError(expectedRespCode []int, resp *http.Response) Error {
 		response:             resp,
 	}
 }
+
+// NewErrorX wraps all details we get during screening or checkr requests
+func NewErrorX(id string, st ScreenType, err error, expectedRespCode int, resp *http.Response) *ErrorX {
```

Comment:
@chandradeepak just a nit pick but ideally, we name the error after the package. So this would be `CheckrError`.

Reply:
yeah @magicalbanana  go lint was saying it would stutter who ever uess it like checkr. CheckrError